refinery/core/api.py

```diff
@@ -498,6 +498,13 @@ class Meta:
         }
 
     def dehydrate(self, bundle):
+        version_details = bundle.obj.get_version_details()
+
+        if version_details is None:
```

Comment:
Its very possible to have an "incomplete" DataSet at that point:

`dehydrate()` is hit before `obj_get()` and `obj_get_list()`
As far as TastyPie cares, its queryset includes: `DataSet.objects.all()` so "Incomplete" DataSets are always included in the response.

I only found one example in our codebase so far, but I'm sure we are doing things like:
```
ds = DataSet.objects.create(name="DataSet")
####
# Some code happens here and a user hits `/api/v1/datasets` during this time
###

# Other objects are then associated 
i = Investigation.objects.create()
s = Study.objects.create(investigation=i)
i_l = InvestigationLink.objects.create(
        data_set=dataset,
        investigation=i,
        version=1,
        date=datetime.now()
    )
ds.save() # DataSet is "complete"
```